spectator/meter/__init__.py

```diff
@@ -12,3 +14,22 @@ def __init__(self, meter_id: MeterId, writer: WriterUnion, meter_type_symbol: st
 
     def writer(self) -> WriterUnion:
         return self._writer
+
+    def update(self, value: Union[c_uint64, float, int]) -> None:
+        """General-purpose meter update method, to assist with meta-programming."""
+        if self._meter_type_symbol == "c" and value > 0:
+            # counter
+            line = f"{self._meter_type_symbol}:{self._id.spectatord_id}:{value}"
+            self._writer.write(line)
+        elif self._meter_type_symbol in ["d", "D", "t", "T"] and value >= 0:
+            # dist_summary, percentile_dist_summary, percentile_timer, timer
+            line = f"{self._meter_type_symbol}:{self._id.spectatord_id}:{value}"
+            self._writer.write(line)
+        elif self._meter_type_symbol == "U" and isinstance(value, c_uint64):
+            # monotonic_counter_uint
+            line = f"{self._meter_type_symbol}:{self._id.spectatord_id}:{value.value}"
+            self._writer.write(line)
+        else:
+            # age_gauge, gauge, max_gauge, monotonic_counter
+            line = f"{self._meter_type_symbol}:{self._id.spectatord_id}:{value}"
+            self._writer.write(line)
```

Comment:
I'm a little unclear on how the `else` is different from the other sections. 

If someone calls `update(-5)` and the meter is a `dist_summary` the `elif` is fail but the `else` will end up publishing the negative value for a `dist_summary`.

Reply:
Good point! Will fix.